pkg/test/runner/runner.go

```diff
@@ -79,7 +79,19 @@ func RunAllWithObjectCreated(ctx context.Context, t *testing.T, mgr manager.Mana
 
 func RunAllWithDependenciesCreatedButNotObject(ctx context.Context, t *testing.T, mgr manager.Manager, shouldRunFunc ShouldRunFunc, testCaseFunc TestCaseFunc) {
 	testFunc := func(ctx context.Context, t *testing.T, testContext TestContext, sysContext SystemContext) {
-		dependencyCleanup := sysContext.Reconciler.CreateAndReconcile(ctx, testContext.DependencyUnstructs, testreconciler.CleanupPolicyAlways, testContext.CreateUnstruct)
+		cleanupFuncs := make([]func(), 0, len(testContext.DependencyUnstructs))
```

Comment:
I assume you'll also use ReconcileGate here?

Reply:
I think so? `sysContext.Reconciler.Reconcile` will determine the correct reconciler to use.

Reply:
Ah, the reason I asked was that I was not sure why it is refactored here. Then I realized that you replaced `ReconcileIfManagedByKCC()` with `Reconcile()`. This change looks good to me.